### PR TITLE
Higher bonds in Replicator and challenge low incorrect bonds

### PR DESCRIPTION
**Pull request by kongzii**
Screenshot by Dropbox Capture

### CODE REVIEW

File: prediction_market_agent/agents/replicate_to_omen_agent/omen_resolve_replicated.py [hunk 1]
```diff
@@ -36,37 +44,52 @@ class FinalizeAndResolveResult(BaseModel):
...
     created_opened_markets = OmenSubgraphHandler().get_omen_binary_markets(
         limit=None,
         creator=public_key,
-        opened_before=before,
-        finalized=False,
+        # We need markets already opened for answers.
+        opened_before=now,
+        # With a little bandwidth for the market to be finalized,
+        # so we have time for processing it without erroring out at the end.
+        finalized_after=now + timedelta(minutes=30),
```
Comment (kongzii) [inline prediction_market_agent/agents/replicate_to_omen_agent/omen_resolve_replicated.py:63]: Before, we queried only Replicator's markest without any response. Now we query all Replicator markets that can still be answered.

File: prediction_market_agent/agents/replicate_to_omen_agent/omen_resolve_replicated.py [hunk 2]
```diff
@@ -86,6 +109,56 @@ def omen_finalize_and_resolve_and_claim_back_all_markets_based_on_others_tx(
...
+            == possible_resolution
+        ):
+            logger.info(
+                f"Market {market.url=} already answered with {possible_resolution=}. Skipping."
+            )
+            continue
+
+        if any(
+            response.bond_xdai >= realitio_bond / 2 for response in existing_responses
+        ):
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent/agents/replicate_to_omen_agent/omen_resolve_replicated.py:148]: But `realitio_bond` is the numerator, so you can't get divide by zero here!

File: prediction_market_agent/agents/replicate_to_omen_agent/deploy.py [hunk 3]
```diff
@@ -64,7 +65,9 @@ def replicate(self, settings: ReplicateSettings) -> None:
         logger.info(
             f"Finalising, resolving and claiming back xDai from existing markets replicated by {keys.bet_from_address}."
         )
-        omen_finalize_and_resolve_and_claim_back_all_markets_based_on_others_tx(keys)
+        omen_finalize_and_resolve_and_claim_back_all_markets_based_on_others_tx(
+            keys, realitio_bond=REPLICATOR_BOND
+        )
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent/agents/replicate_to_omen_agent/deploy.py:70]: This is incorrect!

File: prediction_market_agent/agents/replicate_to_omen_agent/omen_resolve_replicated.py [hunk 4]
```diff
@@ -86,6 +109,56 @@
...
+            logger.info(
+                f"Market {market.url=} already asnwered with bond > {realitio_bond} / 2. Skipping."
```
[the remark this answers is not in the document]
Comment (evangriffiths) [reply prediction_market_agent/agents/replicate_to_omen_agent/omen_resolve_replicated.py:150]: Good spot!

**Review by evangriffiths — APPROVED** [review]

File: prediction_market_agent/agents/replicate_to_omen_agent/omen_resolve_replicated.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kongzii) [reply prediction_market_agent/agents/replicate_to_omen_agent/omen_resolve_replicated.py:148]: Thanks for standing up for me! 😄

**Pull request merged**